src/worker/activities.rs

```diff
@@ -197,6 +207,7 @@ impl WorkerActivityTasks {
             self.heartbeat_manager.evict(task_token.clone());
             let known_not_found = act_info.known_not_found;
             drop(act_info); // TODO: Get rid of dashmap. If we hold ref across await, bad stuff.
+            self.complete_notify.notify_waiters();
```

Comment:
(or with a oneshot channel)
```suggestion
            if self.outstanding_activity_tasks.is_empty() {
              self.complete_notify.notify_one();
            }
```